internal/archive/archive.go

```diff
@@ -157,13 +194,30 @@ func openUbuntu(options *Options) (Archive, error) {
 	if len(options.Version) == 0 {
 		return nil, fmt.Errorf("archive options missing version")
 	}
+	if options.Pro != "" {
+		if _, ok := proArchiveInfo[options.Pro]; !ok {
+			return nil, fmt.Errorf("invalid pro value: %q", options.Pro)
+		}
+	}
+
+	baseURL := archiveURL(options.Pro, options.Arch)
+	var creds *credentials
+	if options.Pro != "" {
+		var err error
+		creds, err = findCredentials(baseURL)
```

Comment:
Optionally, this might go into archiveURL as well and be returned as another result. Chances are every time we get a pro URL we'll need the credentials.

Reply:
Agreed, tried that and it looks better.

Reply:
Done in https://github.com/canonical/chisel/pull/167/commits/3d457a9f712677332fc7fec6d878962671d4ed3d.